blaze/expr/collections.py

```diff
@@ -424,10 +424,36 @@ def join(lhs, rhs, on_left=None, on_right=None, how='inner'):
 
 join.__doc__ = Join.__doc__
 
+class IsIn(Expr):
+    """
+    """
+    __slots__ = '_hash', '_child', '_key'
+
+    def __init__(self, child, key):
+        super(IsIn, self).__init__(child, frozenset(key))
+
+    @property
+    def dshape(self):
+        """ self._child.dshape * number of matched results
+        """
+        return DataShape(*(self._child.shape + (datashape.bool_,)))
```

Comment:
I'm basically copying what your telling me here. What does `dshape` need to do? And once I know this, I guess I can figure out how to express what I want by reading the datashape docs?